client/components/ImageViewer.vue

```diff
@@ -56,30 +76,62 @@ export default {
 		},
 	},
 	watch: {
-		link() {
+		link(newLink, oldLink) {
 			// TODO: history.pushState
-			if (this.link === null) {
+			if (newLink === null) {
+				this.$root.$off("escapekey", this.closeViewer);
+				this.$root.$off("resize", this.correctPosition);
+				Mousetrap.unbind("left", this.previous);
+				Mousetrap.unbind("right", this.next);
 				return;
 			}
 
-			this.$root.$on("resize", this.correctPosition);
+			this.setPrevNextImages();
+
+			if (!oldLink) {
+				this.$root.$on("escapekey", this.closeViewer);
+				this.$root.$on("resize", this.correctPosition);
+				Mousetrap.bind("left", this.previous);
+				Mousetrap.bind("right", this.next);
+			}
 		},
 	},
-	mounted() {
```

Comment:
Failed rebase?

Reply:
I think this is covered by the watch, or is it?

Reply:
Oh this PR includes a commit which is already on master. Can you rebase it cleanly?

Reply:
I didn't realize my commit wasn't on master, so this is fine.